eng/pipelines/maui-release.yml

```diff
@@ -190,4 +196,85 @@ extends:
             os: ${{ parameters.VM_IMAGE_HOST.os }}     
             scanArtifacts: ['${{ parameters.PackPlatform.binariesArtifact }}']
 
-        
+        - ${{ if eq(variables['Build.Reason'], 'Schedule')) }}:
+          - stage: nightly
+            displayName: Nightly
+            dependsOn: ['sdk_insertion']
+            jobs:
+            - job: push_nightly
+              displayName: Push to nightly feed
+              pool: ${{ parameters.VM_IMAGE_HOST }}
+              workspace:
+                clean: all
+              timeoutInMinutes: 60
+              steps:
+              - checkout: none
+
+              - task: UseDotNet@2
+                displayName: Use .NET 8.x
+                inputs:
+                  version: 8.x
+
+              - task: AzureCLI@2
+                displayName: Update service connection with managed identity
+                inputs:
+                  azureSubscription: maui-nightly-feed-mi
+                  scriptType: ps
+                  scriptLocation: inlineScript
+                  inlineScript: |
+                    $accessToken = az account get-access-token --query accessToken --resource $(MauiNightlyFeedMIResourceId) -o tsv
+                    Write-Host "##vso[task.setsecret]$accessToken"
+                    Write-Host "Overwriting authorization token for the maui-nightly-feed service connection"
+                    # https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#setendpoint-modify-a-service-connection-field
+                    Write-Host "##vso[task.setendpoint id=$(MauiNightlyFeedServiceConnectionId);field=authParameter;key=apitoken]${accessToken}"
```

Comment:
Nightly variables are set in the build definition: https://dev.azure.com/devdiv/DevDiv/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=15722&view=Tab_Variables